### PR TITLE
Added `pvcviewer-controller` rock

### PR DESCRIPTION
**Pull request by TakoB222**
### Description
-----
- Add rockcraft.yaml
- Add tests
- Add tox

### CODE REVIEW

File: pvcviewer-controller/rockcraft.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+description: |
+  This image is used as part of the Charmed Kubeflow product. 
+  PVCViewers enable users to open a filebrowser on arbitrary persistent volume claims, letting them inspect, download, upload and manipulate data.
+version: "1.9.0"
+license: Apache-2.0
+base: ubuntu@22.04
+
+platforms:
+  amd64:
+
```
Comment (misohu) [inline pvcviewer-controller/rockcraft.yaml:13]: Missing run-user as here https://github.com/canonical/kserve-rocks/blob/main/agent/rockcraft.yaml

File: pvcviewer-controller/rockcraft.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+services:
+  pvcviewer-controller:
+    override: merge
```
Comment (misohu) [inline pvcviewer-controller/rockcraft.yaml:16]: Why merge ? we use replace as default if merge is not needed

File: pvcviewer-controller/rockcraft.yaml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    command: "pvc-viewer"
+    startup: enabled
+    user: ubuntu
```
Comment (misohu) [inline pvcviewer-controller/rockcraft.yaml:19]: We dont use user ubuntu in services example rock https://github.com/canonical/kserve-rocks/blob/main/agent/rockcraft.yaml

File: pvcviewer-controller/rockcraft.yaml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+parts:
+  security-team-requirement:
```
Comment (misohu) [inline pvcviewer-controller/rockcraft.yaml:22]: Update security requirement, example: https://github.com/canonical/kserve-rocks/blob/main/agent/rockcraft.yaml#L21

File: pvcviewer-controller/rockcraft.yaml [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    plugin: nil
+    override-build: |
+      mkdir -p ${CRAFT_PART_INSTALL}/usr/share/rocks
+      (echo "# os-release" && cat /etc/os-release && echo "# dpkg-query" && dpkg-query -f '${db:Status-Abbrev},${binary:Package},${Version},${source:Package},${Source:Version}\n' -W) > ${CRAFT_PART_INSTALL}/usr/share/rocks/dpkg.query
+
+  pvcviewer-controller:
+    plugin: go
+    build-snaps:
+      - go/1.21/stable
```
Comment (misohu) [inline pvcviewer-controller/rockcraft.yaml:31]: upstream uses 1.22.2 https://github.com/kubeflow/kubeflow/blob/v1.9.0/components/pvcviewer-controller/Dockerfile#L2

File: pvcviewer-controller/rockcraft.yaml [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    user: ubuntu
+
+parts:
+  security-team-requirement:
+    plugin: nil
+    override-build: |
+      mkdir -p ${CRAFT_PART_INSTALL}/usr/share/rocks
+      (echo "# os-release" && cat /etc/os-release && echo "# dpkg-query" && dpkg-query -f '${db:Status-Abbrev},${binary:Package},${Version},${source:Package},${Source:Version}\n' -W) > ${CRAFT_PART_INSTALL}/usr/share/rocks/dpkg.query
+
+  pvcviewer-controller:
```
Comment (misohu) [inline pvcviewer-controller/rockcraft.yaml:28]: To stay consistent I recomment building like this https://github.com/canonical/kubeflow-rocks/blob/main/admission-webhook/rockcraft.yaml#L41 with proper environment

File: pvcviewer-controller/rockcraft.yaml [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+license: Apache-2.0
+base: ubuntu@22.04
+
+platforms:
+  amd64:
+
+services:
+  pvcviewer-controller:
+    override: merge
+    command: "pvc-viewer"
```
Comment (misohu) [inline pvcviewer-controller/rockcraft.yaml:17]: Upstream command is `/manager` not pvc-viewer https://github.com/kubeflow/kubeflow/blob/v1.9.0/components/pvcviewer-controller/Dockerfile#L28

File: pvcviewer-controller/tests/test_rock.py [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+    subprocess.run(
+        [
+            "docker",
+            "run",
+            "--rm",
+            LOCAL_ROCK_IMAGE,
+            "exec",
+            "ls",
+            "-la",
+            "/bin/pvc-viewer",
```
Comment (misohu) [inline pvcviewer-controller/tests/test_rock.py:52]: Change the test to check for /manager as upstream is using that https://github.com/kubeflow/kubeflow/blob/v1.9.0/components/pvcviewer-controller/Dockerfile#L28

File: pvcviewer-controller/rockcraft.yaml (hunk 2 above)
Comment (TakoB222) [reply pvcviewer-controller/rockcraft.yaml:16]: Yeah, missed this, will fix

**Comment by misohu** [comment]
Hi @TakoB222 before we proceed we approval of this PR can you please make sure the rock is working with existing charm. You can do it by running [integration tests](https://github.com/canonical/pvcviewer-operator/blob/main/.github/workflows/integrate.yaml#L59) from the pvc viewer repo. 

1. Deploy microk8s 1.29-strict/stable
2. Connect it with juju 3.4/stable 
3. Create `kubeflow model`
(if you will have problems with the juju setup please try [this ](https://charmed-kubeflow.io/docs/get-started-with-charmed-kubeflow)step by step guide until model deployment )
4. publish rock to your dockerhub e.g. `tacko/pvc-viewer`
5. In the charm repo replace [this ](https://github.com/canonical/pvcviewer-operator/blob/main/metadata.yaml#L21) line with your rock image from your dockerhub.
6. Run the tox integration test and keep the model `tox -vve integration -- --model kubeflow --keep-model`

All the tests should be green. After successfully running the tests please can you provide the output of `juju status`.
When we develop in our team that is the way how we test that the rock is ready.

**Comment by DnPlas** [comment]
Closing this as a duplicate of #154

**Pull request closed without merging**